Inc/stm32f446xx_i2c.h

```diff
@@ -11,34 +11,46 @@
 #include<stdint.h>
 #include"stm32f446xx.h"
 
-typedef struct{
 
+class I2C_Config_t:{
+
+
+  private:
 	__vo uint8_t I2C_SCLSpeed	;							//CAN		BE 	ANYONE	OF	@I2C_SCLSpeed
 	__vo uint8_t I2C_DeviceAddress	;							//CAN		BE 	ANYONE	OF	@I2C_DeviceAddress
 	__vo uint8_t I2C_ACKControl	;							//CAN		BE 	ANYONE	OF	@I2C_ACKControl
 	__vo uint8_t I2C_FMDutyCycle	;							//CAN		BE 	ANYONE	OF	@I2C_FMDutyCycle
 
-}I2C_Config_t;
+I2C_Config_t(): I2C_SCLSpeed(0),I2C_DeviceAddress(0),I2C_ACKControl(0),I2C_FMDutyCycle(0){};
+
+protected:
+  void I2C_Config(uint8_t speed=0,uint8_t deviceaddr=0, uint8_t ackcontrol=0,uint8_t fmdutycycle=0)
+  {
+      I2C_SCLSpeed=speed;
+	  I2C_DeviceAddress= deviceaddr;
+	  I2C_DeviceAddress= ackcontrol;
+	  I2C_FMDutyCycle=fmdutycycle;
+
+  }
+
+
+};
 
 
 /*
  * HANDLE		STRUCTURE		FOR 		I2C 		PERIPHERAL
  */
-typedef struct{
+class I2C_Handle_t:public I2C_Config_t
+{
 
-	I2C_RegDef_t	*pI2Cx;
-	I2C_Config_t	I2C_Config;
 
-	uint8_t			*pTxBuffer;			// !<	  To store the app. Tx buffer address 	>
```

Comment:
Add these also

Reply:
till
uint8_t			Sr	;						// !<	  To store Repeated start value	>

right?